### ISSUE TITLE
The problem of the enhanced results while using different loss functions?

### ISSUE THREAD

**Issue by YoungJay0612** [facebookresearch/denoiser#33]
Hello, When I train the demucs model with l1-loss，the enhanced result is very well, and the speech sense is always satisfactory 。But when I change the loss function from l1-loss to l1-loss+0.1* multi-resolution STFT loss， I just changed the stft_loss in the configuration file config.yaml to be true in the configuration file。But the enhanced speech introduced some noise.This is the example file https://github.com/YoungJay0612/demucs-enhanced-result. I have thought a lot and have no idea now. 
Hopefully to hear from you.

**Comment by adiyoss** [comment]
Hi @YoungJay0612 ,
As a general statement, from our experience the stft-loss won't always improve results, it depends on the dataset. 
What you can try to do is: 
1) Since the value range the L1 loss is different than the `stft-loss`, I suggest grid-search over several coefficient values, they are called: `stft_sc_factor`, and `stft_mag_factor` on the config file. 
2) the `stft-loss` is composed of two terms, spectral-convergence (sc) and log magnitude (mag), I suggest to explore L1 + log magnitude loss only, you can do that by setting `stft_sc_factor` to zero.

**Comment by YoungJay0612** [comment]
Sorry to tell you that the question is still exist. And I upload four pictures of the processed wav(see the results links). FIG.2 is the result of the Demucs with l1-loss , and  FIG. 1 is the result of a certain magnification of FIG. 2. As we can see ,there is no multiple single frequency components. FIG.3 and FIG.4 is the corresponding result of the same module but with L1-loss + 0.1*sm-loss, FIG.3 is the certain magnification version of FIG. 4. And the background multiple single frequency components is  obvisious. Is this because Demucs network is time domain signal processing and STFT is frequency domain loss?
Hopefully to hear from you!
results: [https://github.com/YoungJay0612/demucs-result](url)

**Comment by adiyoss** [comment]
Hi @YoungJay0612,
Regarding your question, the STFT is done after the model predicts the waveform, so in both cases, the model acts in the same way.  
Since the model is working fine when you are optimizing with the L1 loss, my guess is that the STFT loss is taking control over the optimization and probably focuses on some artifacts rather than the actual signal. 
Did you try to use the L1 + magnitude loss only? Did you try to use other (smaller/bigger) coefficient values than 0.1?

**Comment by YoungJay0612** [comment]
As you mentioned, I have done some comparative experiments including L1 loss, L1 + stft loss, L1 + 0.5*stft loss and L1 + 0.01 magnitude loss, etc. Unfortunately, the phenomenon occurs in every situation except L1, but there is a difference in severity.  The higher the weighted coefficient of stft loss, the more obvious the phenomenon is. 
Similarly, I also try to  combine Phone Fortified Perceptual Loss and L1 loss which was described in [https://github.com/aleXiehta/PhoneFortifiedPerceptualLoss](url) to improve the perceptual quality of the enhanced speech.  And according to the results, the performance of the demucs module is improved significantly both in noise suppression and speech retention. But the problem remains the same, there is also some multiple single frequency components in the background which is the same to the L1 + stft loss. It disturbs me several weeks. 
Hpoefully to hear from you again!

**Comment by jhkonan** [comment]
This seems to still be an issue...

**Comment by YunyangZeng** [comment]
We observed similar single frequency noise while fine-tuning the model using both the L1 loss and our loss. The pesq and stoi were great but the single frequency noise is obvious in the spectrogram...
